lib/path.rb

```diff
@@ -0,0 +1,66 @@
+class Metis
+    class Path
+      # Assumes Metis file paths are in the form
+      #    metis://<project>/<bucket>/<folder path>/<file name>
+      # Splitting the above produces
+      #   ["metis", "", "<project>", "<bucket>", "<folder path>" ... "file name"]
+
+      FILENAME_MATCH=/[^<>:;,?"*\|\/\x00-\x1f]+/x
+
+      FILEPATH_MATCH=%r!
+        \A
+          metis:\/\/
+          (?<project_name>(#{FILENAME_MATCH.source}))
+          /
+          (?<bucket_name>(#{FILENAME_MATCH.source}))
+          /
+          (?<file_path>(#{FILENAME_MATCH.source}).*)
+        \z
+      !x
+
+      attr_reader :path
+      def initialize(path)
+        @path = path
+      end
+
+      def self.extract_bucket_name_from_path(path)
+        return FILEPATH_MATCH.match(path)[:bucket_name]
```

Comment:
Does this have value as a class method? Could these just be folded into `#bucket_name`, etc?

Reply:
Only have it as a class method because I was using it in the `metis_controller` methods. Can remove it after I refactor those.